[Bug][Azure DevOps] Entities Column Missing

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.


### What happened

When standing up a new v0.18.0-beta1 deployment and trying to create scope config, it errors indicating that the entities column does not exist on the _tool_azuredevops_gitrepositoryconfigs table.

```
time="2023-07-18 16:55:18" level=info msg="/app/impls/dalgorm/dalgorm.go:196 Error 1054 (42S22): Unknown column 'entities' in 'field list'
[1.704ms] [rows:0] INSERT INTO `_tool_azuredevops_gitrepositoryconfigs` (`created_at`,`updated_at`,`entities`,`connection_id`,`name`,`refdiff`,`deployment_pattern`,`production_pattern`) VALUES ('2023-07-18 16:55:18.926','2023-07-18 16:55:18.926','["CODE","CODEREVIEW","CROSS","CICD"]',1,'DevSecOps Scope Config',NULL,'(deploy|push-image)','')"
```

### What do you expect to happen

The column to exist. 

### How to reproduce

Stand up a new v0.18.0-beta1 environment. The 

### Anything else

_No response_

### Version

v0.18.0-beta1

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/backend/python/plugins/azuredevops/azuredevops/models.py
+++ b/backend/python/plugins/azuredevops/azuredevops/models.py
@@ -145,6 +145,10 @@ def add_build_id_as_job_primary_key(b: MigrationScriptBuilder):
 @migration(20230606165630)
 def rename_tx_rule_table_to_scope_config(b: MigrationScriptBuilder):
     b.rename_table('_tool_azuredevops_azuredevopstransformationrules', GitRepositoryConfig.__tablename__)
+
+
+@migration(20230607165630, name="add entities column to gitrepositoryconfig table")
+def add_entities_column_to_scope_config(b: MigrationScriptBuilder):
     b.add_column(GitRepositoryConfig.__tablename__, 'entities', 'json')
 
 

--- a/backend/server/services/remote/models/migration.go
+++ b/backend/server/services/remote/models/migration.go
@@ -56,7 +56,10 @@ type AddColumnOperation struct {
 
 func (o AddColumnOperation) Execute(dal dal.Dal) errors.Error {
 	if dal.HasColumn(o.Table, o.Column) {
-		return dal.DropColumns(o.Table, o.Column)
+		err := dal.DropColumns(o.Table, o.Column)
+		if err != nil {
+			return err
+		}
 	}
 	return dal.AddColumn(o.Table, o.Column, o.ColumnType)
 }
@@ -162,9 +165,9 @@ func (s *RemoteMigrationScript) UnmarshalJSON(data []byte) error {
 }
 
 func (s *RemoteMigrationScript) Up(basicRes context.BasicRes) errors.Error {
-	dal := basicRes.GetDal()
+	db := basicRes.GetDal()
 	for _, operation := range s.operations {
-		err := operation.Execute(dal)
+		err := operation.Execute(db)
 		if err != nil {
 			return err
 		}

--- a/backend/server/services/remote/plugin/plugin_impl.go
+++ b/backend/server/services/remote/plugin/plugin_impl.go
@@ -231,26 +231,33 @@ func (p *remotePluginImpl) ApiResources() map[string]map[string]plugin.ApiResour
 }
 
 func (p *remotePluginImpl) RunAutoMigrations() errors.Error {
-	db := basicRes.GetDal()
-	err := api.CallDB(db.AutoMigrate, p.connectionTabler.New())
+	err := p.createTable(p.connectionTabler.New())
 	if err != nil {
 		return err
 	}
-	err = api.CallDB(db.AutoMigrate, p.scopeTabler.New())
+	err = p.createTable(p.scopeTabler.New())
 	if err != nil {
 		return err
 	}
-	err = api.CallDB(db.AutoMigrate, p.scopeConfigTabler.New())
+	err = p.createTable(p.scopeConfigTabler.New())
 	if err != nil {
 		return err
 	}
 	for _, toolModelTabler := range p.toolModelTablers {
-		err = api.CallDB(db.AutoMigrate, toolModelTabler.New())
+		err = p.createTable(toolModelTabler.New())
 		if err != nil {
 			return err
 		}
 	}
 	return nil
+}
+
+func (p *remotePluginImpl) createTable(tbl coreModels.DynamicTabler) errors.Error {
+	db := basicRes.GetDal()
+	if db.HasTable(tbl.TableName()) {
+		return nil
+	}
+	return api.CallDB(db.AutoMigrate, tbl)
 }
 
 func (p *remotePluginImpl) OpenApiSpec() string {

